extensions/servlet/runtime/src/main/java/org/apache/camel/quarkus/servlet/runtime/CamelServletConfig.java

```diff
@@ -87,6 +92,37 @@ public String getEffectiveServletName(final String key) {
             return DEFAULT_SERVLET_NAME.equals(servletName) ? key : servletName;
         }
 
+        /**
+         * Servlet multipart request configuration.
+         */
+        @ConfigGroup
+        public static class MultipartConfig {
+            /**
+             * An absolute path to a directory on the file system to store files temporarily while the parts are
+             * processed or when the size of the file exceeds the specified file-size-threshold configuration value.
+             */
+            @ConfigItem(defaultValue = "${java.io.tmpdir}")
+            public String location;
+
+            /**
+             * The maximum size allowed in bytes for uploaded files. The default size (-1) allows an unlimited size.
+             */
+            @ConfigItem(defaultValue = "-1")
+            public Optional<Long> maxFileSize;
+
+            /**
+             * The maximum size allowed in bytes for a multipart/form-data request. The default size (-1) allows an unlimited
+             * size.
+             */
+            @ConfigItem(defaultValue = "-1")
+            public Optional<Long> maxRequestSize;
+
+            /**
+             * The file size in bytes after which the file will be temporarily stored on disk.
+             */
+            @ConfigItem(defaultValue = "0")
+            public Optional<Integer> fileSizeThreshold;
```

Comment:
I wonder whether all these need be optional given that they all have defaults? The optionals will always be non-empty. 

I also wonder why fileSizeThreshold is int while the others are long?

Reply:
>  The optionals will always be non-empty

I can remove the `Optional`.

> I also wonder why fileSizeThreshold is int while the others are long

See [MultipartConfigElement](https://jakarta.ee/specifications/servlet/5.0/apidocs/jakarta/servlet/multipartconfigelement).